cli/azd/cmd/infra_create.go

```diff
@@ -272,25 +276,78 @@ func (ica *infraCreateAction) Run(ctx context.Context, cmd *cobra.Command, args
 	resourceManager := infra.NewAzureResourceManager(azCli)
 
-	operations, err := resourceManager.GetDeploymentResourceOperations(ctx, env.GetSubscriptionId(), env.GetEnvName())
+	resOps, err := resourceManager.GetDeploymentResourceOperations(ctx, env.GetSubscriptionId(), env.GetEnvName())
 	if err != nil {
 		// Status display is best-effort activity.
 		return
 	}
 
+	operations := *resOps
 	succeededCount := 0
+	newlyDeployedResources := []azureutil.DeployedAzureResource{}
+	loggedResources := *loggedResourcesPtr
 
-	for _, resourceOperation := range *operations {
+	for _, resourceOperation := range operations {
 		if resourceOperation.Properties.ProvisioningState == "Succeeded" {
 			succeededCount++
+
+			if !loggedResources[resourceOperation.Properties.TargetResource.Id] {
+				newlyDeployedResources = append(newlyDeployedResources, azureutil.DeployedAzureResource{
+					Id:                resourceOperation.Properties.TargetResource.Id,
+					ResourceName:      resourceOperation.Properties.TargetResource.ResourceName,
+					ResourceType:      resourceOperation.Properties.TargetResource.ResourceType,
+					DeployedTimestamp: resourceOperation.Properties.Timestamp,
+				})
+			}
+		}
+	}
+
+	sort.Sort(azureutil.DeployedAzureResourceByTimestamp(newlyDeployedResources))
+
+	for i := range newlyDeployedResources {
+		if newlyDeployedResources[i].ResourceType == string(infra.AzureResourceTypeWebSite) {
+			// Web apps have different kinds of resources sharing the same resource type 'Microsoft.Web/sites', i.e. Function app vs. App service
+			// We resolve it by querying the properties of the ARM resource.
+			resourceTypeName, err := resourceManager.GetWebAppResourceTypeDisplayName(ctx, env.GetSubscriptionId(), newlyDeployedResources[i].Id)
+
+			if err != nil {
+				// Best effort -- use static translation 'Web App' if we could not resolve the exact resource type
+				newlyDeployedResources[i].ResourceTypeDisplayName = infra.GetResourceTypeDisplayName(infra.AzureResourceType(newlyDeployedResources[i].ResourceType))
+			} else {
+				newlyDeployedResources[i].ResourceTypeDisplayName = resourceTypeName
+			}
+		} else {
+			resourceTypeName := infra.GetResourceTypeDisplayName(infra.AzureResourceType(newlyDeployedResources[i].ResourceType))
+
+			if resourceTypeName != "" {
+				newlyDeployedResources[i].ResourceTypeDisplayName = resourceTypeName
+			}
+		}
+	}
+
+	for _, newResource := range newlyDeployedResources {
+		// Don't log resource types that are not known to us.
+		// There are two cases:
+		// 1. Resource types that are sub-components. i.e., application settings: "Microsoft.Web/sites/config"
+		// 2. Azure resources that we do not have a translation of the resource type for.
+		if newResource.ResourceTypeDisplayName != "" {
+			spinnerUpdater.LogMessage(fmt.Sprintf("Created - %s: %s - %s", newResource.ResourceTypeDisplayName, newResource.ResourceName, newResource.DeployedTimestamp.Format(time.ANSIC)))
```

Comment:
@Austinauth to review the final progress log format we want

Reply:
FWIW - I'd consider putting the date first in the line (if we are going to have it all, perhaps if we start reporting operation durations we don't actually need it) and perhaps pick a time format that's targeted more at humans running this tool on the CLI (the full date is maybe uninteresting in this case?)

Reply:
That's a keen observation. Putting the date first in the line makes total sense. Austin had provided the following sketch:

![MicrosoftTeams-image](https://user-images.githubusercontent.com/2322434/180583621-78d6a8b8-f2fd-47d1-8e1b-49ea27a3ae8c.png)

Given that the target audience here are devs and we desire a logging format, I think the full date time does make sense.